binder/binder.c

```diff
@@ -4109,7 +4109,9 @@ static int binder_wait_for_work(struct binder_thread *thread,
 	struct binder_proc *proc = thread->proc;
 	int ret = 0;
 
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6,0,0)
 	freezer_do_not_count();
+#endif
 	binder_inner_proc_lock(proc);
 	for (;;) {
 		prepare_to_wait(&thread->wait, &wait, TASK_INTERRUPTIBLE);
```

Comment:
Sorry for the late code review :-( .. I was extemely busy in the last weeks moving from one place to another.

I think your patch is not fully correct. When you check the changes made to binder.c [in this commit](https://github.com/torvalds/linux/commit/f5d39b020809146cc28e6e73369bf8065e0310aa#diff-dc675d837f5054b5e96c5fc9fcb77ad08b94607f88b692930132614aa83613b2), then you will see that prepare_to_wait() is now called with TASK_FREEZABLE flag. You should also add this flag here.

Reply:
Can confirm, Waydroid and Apple Music start on `6.1.8-arch1-1` with `TASK_INTERRUPTIBLE|TASK_FREEZABLE`.

For reference, I have `sys.use_memfd=true` and am using LineageOS 18.1 with Gapps.

Reply:
I have added TASK_FREEZABLE flag, thanks for your advice for fixing this resulted by my careless.